src/storage.cc

```diff
@@ -0,0 +1,386 @@
+  cyclus::toolkit::CommodityProducer::SetCapacity(commod, capacity);
+  cyclus::toolkit::CommodityProducer::SetCost(commod, cost);
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void Storage::EnterNotify() {
+  Facility::EnterNotify();
+
+  using cyclus::toolkit::Commodity;
+  Commodity commod = Commodity(out_commod);
+  cyclus::toolkit::CommodityProducer::Add(commod);
+  cyclus::toolkit::CommodityProducer::SetCapacity(commod, capacity);
+  cyclus::toolkit::CommodityProducer::SetCost(commod, cost);
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+std::string Storage::str() {
+  std::stringstream ss;
+  std::string ans;
+  if (cyclus::toolkit::CommodityProducer::
+      Produces(cyclus::toolkit::Commodity(out_commod_()))){
+    ans = "yes";
+  } else {
+    ans = "no";
+  }
+  ss << cyclus::Facility::str();
+  ss << " has facility parameters {" << "\n"
+     << "     Input Commodity = " << in_commod_() << ",\n"
+     << "     Output Commodity = " << out_commod_() << ",\n"
+     << "     Process Time = " << process_time_() << ",\n"
+     << "     Capacity = " << capacity_() << ",\n"
+     << " commod producer members: " << " produces "
+     << out_commod << "?:" << ans
+     << " capacity: " << cyclus::toolkit::CommodityProducer::Capacity(out_commod_())
+     << " cost: " << cyclus::toolkit::CommodityProducer::Cost(out_commod_())
+     << "'}";
+  return ss.str();
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void Storage::Tick() { 
+  LOG(cyclus::LEV_INFO3, "ComCnv") << prototype() << " is ticking {";
+
+  if (current_capacity() > cyclus::eps()) {
+    LOG(cyclus::LEV_INFO4, "ComCnv") << " has capacity for " << current_capacity()
+                                       << " kg of " << in_commod
+                                       << " recipe: " << in_recipe << ".";
+  }
+
+  LOG(cyclus::LEV_INFO3, "ComCnv") << "}";
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void Storage::Tock() {
+  LOG(cyclus::LEV_INFO3, "ComCnv") << prototype() << " is tocking {";
+
+  BeginProcessing_(); // place unprocessed inventory into processing
+
+  if( ready() >= 0 || process_time_() == 0 ) {
+    Convert_(capacity_()); // place processing into stocks
+  }
+
+  LOG(cyclus::LEV_INFO3, "ComCnv") << "}";
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+std::set<cyclus::RequestPortfolio<cyclus::Material>::Ptr>
+Storage::GetMatlRequests() {
+  using cyclus::CapacityConstraint;
+  using cyclus::Material;
+  using cyclus::RequestPortfolio;
+  using cyclus::Request;
+
+  std::set<RequestPortfolio<Material>::Ptr> ports;
+
+  RequestPortfolio<Material>::Ptr port(new RequestPortfolio<Material>());
+  Material::Ptr mat = Request_();
+  double amt = mat->quantity();
+
+  if (amt > cyclus::eps()) {
+    //CapacityConstraint<Material> cc(amt);
+    //port->AddConstraint(cc);
```

Comment:
I probably commented this out because I wasn't sure (hadn't sufficiently tested) whether or not adding a capacity constraint was necessary. It needs to be determined whether this is necessary (I suspect it is) and ideally it should be tested to verify that it has the appropriate behavior (that is, it shouldn't request more than its capacity).